src/compiler/config/validate-service-worker.ts

```diff
@@ -56,16 +70,19 @@ export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d
     outputTarget.serviceWorker.swSrc = join(config.rootDir, outputTarget.serviceWorker.swSrc);
   }
 
-  if (!isString(outputTarget.serviceWorker.swDest)) {
-    outputTarget.serviceWorker.swDest = join(outputTarget.appDir, DEFAULT_FILENAME);
-  }
-
-  if (!isAbsolute(outputTarget.serviceWorker.swDest)) {
-    outputTarget.serviceWorker.swDest = join(outputTarget.appDir, outputTarget.serviceWorker.swDest);
+  if (isString(outputTarget.serviceWorker.swDest) && !isAbsolute(outputTarget.serviceWorker.swDest)) {
```

Comment:
If we change the condition above (previous comment), then this `isString` check can go away
```suggestion
  if (!isAbsolute(outputTarget.serviceWorker.swDest)) {
```

Reply:
do you mean the ternary? I don't think that's the case because this line isn't inside of an `if` block which will cause the `.swDest` type to be narrowed.

Also, and this is a bit funky, but the narrowing with `isString` takes it from `swDest?: string | undefined` to `swDest?: string` (i.e. it's still effectively `string | undefined`) so we don't make too much headway there :/

Reply:
Hmm yeah that's a bit confusing. I was thinking the type of `swDest` was narrowed to explicitly a string in lines 53-55 so we would only need to check if that path was absolute or not.